knack/arguments.py

```diff
@@ -87,63 +124,82 @@ def get_cli_argument(self, command, name):
+    def argument(self, argument_dest, arg_type=None, **kwargs):
+        """ Register an argument for the given command scope using a knack.arguments.CLIArgumentType
+
+        :param argument_dest: The destination argument to add this argument type to
+        :type argument_dest: str
+        :param arg_type: The argument type we are registering
+        :type arg_type: knack.arguments.CLIArgumentType
+        :param kwargs: Any argument type overrides that need to be applied
+        """
+        self.command_loader.argument_registry.register_cli_argument(self.command_scope,
+                                                                    argument_dest,
                                                                     arg_type,
                                                                     **kwargs)
 
-    def register_alias(self, argument_name, options_list, **kwargs):
-        self.command_loader.argument_registry.register_cli_argument(self.commmand,
-                                                                    argument_name,
-                                                                    None,
-                                                                    options_list=options_list,
-                                                                    **kwargs)
+    def register(self, argument_dest, options_list, **kwargs):
+        """ Register an argument for the given command scope without a knack.arguments.CLIArgumentType.
+            You add the kwargs you require yourself.
 
-    def register(self, argument_name, options_list, **kwargs):
-        self.command_loader.argument_registry.register_cli_argument(self.commmand,
-                                                                    argument_name,
-                                                                    None,
-                                                                    options_list=options_list,
-                                                                    **kwargs)
+        :param argument_dest: The destination argument to add this argument type to
+        :type argument_dest: str
+        :param options_list: The options that can be supplied for this argument
+        :type options_list: tuple
+        :param kwargs: Any argument type overrides that need to be applied
+        """
+        self.argument(argument_dest, arg_type=None, options_list=options_list, **kwargs)
+
+    def ignore(self, argument_dest):
+        """ Register an argument with type knack.arguments.ignore_type (hidden/ignored)
+
+        :param argument_dest: The destination argument to apply the ignore type to
+        :type argument_dest: str
+        """
+        self.argument(argument_dest, arg_type=ignore_type)
 
     def extra(self, dest, **kwargs):
         '''Register extra parameters for the given command. Typically used to augment auto-command built
         commands to add more parameters than the specific SDK method introspected.
```

Comment:
I'd recommend a copy-paste of the kwargs here as well since this is an end-point that the user would likely want Intellisense for.

Reply:
done.